swagger.yaml

```diff
@@ -1,75 +1,392 @@
+  title: TeamDynamix Wrapper API
+  description: Wrapper API for TeamDynamix to perform some data manipulation.
+  version: '1.0.0'
   license:
     name: GNU Affero General Public License Version 3
     url: http://www.gnu.org/licenses/agpl-3.0.en.html
 externalDocs:
   description: GitHub Repository
-  url: https://github.com/osu-mist/web-api-skeleton
+  url: https://github.com/osu-mist/td-wrapper-api
 host: localhost
 schemes:
-  - http
-basePath: /api/v0
-consumes:
-  - text/plain
+  - https
+basePath: /v1
 produces:
-  - text/plain
+  - application/json
+consumes:
+  - application/json
 paths:
-  /:
+  /teamdynamix/services:
+    get:
+      summary: Get all services
+      description: Get all services with their long descriptions.
+      parameters:
+        - $ref: '#/parameters/pretty'
+        - $ref: '#/parameters/authorization'
+      responses:
+        200:
+          description: Object containing information
+          schema:
+            $ref: '#/definitions/ServicesResultObject'
+        500:
+          description: Internal Server Error
+          schema:
+            $ref: '#/definitions/Errors'
+  /teamdynamix/services/{id}:
     get:
-      summary: Get information
-      description: Build and runtime information about the application.
+      summary: Get a single service by its ID
+      description: Get a single service by its ID.
       parameters:
         - $ref: '#/parameters/pretty'
+        - $ref: '#/parameters/id'
       responses:
         200:
           description: Object containing information
           schema:
-            $ref: '#/definitions/Info'
-
+            $ref: '#/definitions/ServiceResultObject'
+        404:
+          description: Service with the specific ID not found.
+          schema:
+            $ref: '#/definitions/Errors'
+        500:
+          description: Internal Server Error
+          schema:
+            $ref: '#/definitions/Errors'
 parameters:
+  id:
+    name: id
+    in: path
+    required: true
+    description: Unique ID of TD service
+    type: string
+  authorization:
+    name: Authorization
+    in: header
+    type: string
+    required: true
+    description: '"Bearer [token]" where [token] is your OAuth2 access token'
   pretty:
     name: pretty
     in: query
     type: boolean
     required: false
     description: If true, JSON response will be pretty-printed
-
 definitions:
-  Info:
+  ServicesResultObject:
+    properties:
+      data:
+        type: array
+        items:
+          $ref: "#/definitions/ServiceResourceObject"
+  ServiceResultObject:
+    properties:
+      data:
+        properties:
+          id:
+            type: string
+            description: service ID.
+          type:
+            type: string
+            description: service
+          links:
+            $ref: "#/definitions/NullSelfLink"
+          attributes:
+            $ref: "#/definitions/ServiceResourceObject"
+  ServiceResourceObject:
     properties:
+      attachments:
```

Comment:
This object works fine inside a serviceresultobject, but when its in servicesresultobject, its missing the id, type, attributes, etc. I think id, type, and attributes should be in the resource object.